### PR TITLE
feat: Price Feeder in Ojo node

### PR DESCRIPTION
**Pull request by rbajollari**
## Description

closes: #XXXX

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: pricefeeder/price-feeder.example.toml [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+write_timeout = "20s"
+
+[account]
+address = "ojo1zypqa76je7pxsdwkfah6mu9a583sju6xzthge3"
+chain_id = "ojo-testnet"
+validator = "ojovaloper1zypqa76je7pxsdwkfah6mu9a583sju6x6tnq6w"
+
+[keyring]
+backend = "test"
+dir = "/Users/username/.ojo"
```
Comment (adamewozniak) [inline pricefeeder/price-feeder.example.toml:329]:
```suggestion
```

File: pricefeeder/price-feeder.example.toml [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+enable-service-label = true
+enabled = true
+global-labels = [["chain_id", "ojo-devnet-01"]]
+service-name = "price-feeder"
+prometheus-retention-time = 100
+
+[[provider_endpoints]]
+name = "binance"
+rest = "https://api1.binance.com"
+websocket = "stream.binance.com:9443"
```
Comment (adamewozniak) [inline pricefeeder/price-feeder.example.toml:348]:
```suggestion
```

File: pricefeeder/pricefeeder.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+	signal.Notify(sigCh, syscall.SIGTERM)
+	signal.Notify(sigCh, syscall.SIGINT)
+
+	go func() {
+		sig := <-sigCh
+		logger.Info().Str("signal", sig.String()).Msg("caught signal; shutting down...")
+		cancel()
+	}()
+}
+
```
Comment (adamewozniak) [inline pricefeeder/pricefeeder.go:128]:
```suggestion
// startPriceFeeder does xyz
```

File: pricefeeder/pricefeeder.go [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+			return nil
+
+		case err := <-srvErrCh:
+			logger.Error().Err(err).Msg("failed to start price-feeder server")
+			return err
+		}
+	}
+}
+
```
Comment (adamewozniak) [inline pricefeeder/pricefeeder.go:183]:
```suggestion
// startPriceOracle does xyz
```

**Review by adamewozniak — APPROVED** [review]
lgtm, just a couple nits

**Comment by rbajollari** [comment]
@Mergifyio backport main

**Pull request merged**